domain/domain.go

```diff
@@ -0,0 +1,48 @@
+package domain
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"time"
+)
+
+func RunTask(url, task string, timeout time.Duration) error {
```

Comment:
It's usually helpful to have refactors, especially ones that are just code moved from one place to another, be done as a separate PR. Or, at least as a separate commit so it's easier to see what changes were made.